kuma/javascript/src/page.jsx

```diff
@@ -0,0 +1,165 @@
+    }),
+    sidebar: css({
+        flex: 1
+    }),
+    breadcrumbs: css({}),
+    quicklinks: css({})
+};
+
+function Titlebar() {
+    const documentData = useContext(DocumentProvider.context);
+
+    return (
+        documentData && (
+            <Row css={styles.titlebar}>
+                <h1 css={styles.title}>{documentData.title}</h1>
+            </Row>
+        )
+    );
+}
+
+function TOC() {
+    const documentData = useContext(DocumentProvider.context);
+
+    return (
+        documentData && (
+            <Row
+                css={styles.toc}
+                dangerouslySetInnerHTML={{
+                    __html: `<ol>${documentData.tocHTML}</ol>`
+                }}
+            />
+        )
+    );
+}
+
+function Sidebar() {
+    const documentData = useContext(DocumentProvider.context);
+
+    return (
+        documentData && (
+            <div css={styles.sidebar}>
+                <Breadcrumbs />
+                <Quicklinks />
+            </div>
+        )
+    );
+}
+
+function Breadcrumbs() {
+    const documentData = useContext(DocumentProvider.context);
+
+    // The <span> elements below aren't needed except that the stylesheets
+    // are set up to expect them.
+    return (
+        documentData && (
+            <nav className="crumbs" role="navigation">
+                <ol>
+                    {documentData.parents.map(p => (
+                        <li className="crumb" key={p.url}>
+                            <a href={p.url}>
+                                <span>{p.title}</span>
+                            </a>
+                        </li>
+                    ))}
+                    <li className="crumb">
+                        <span>{documentData.title}</span>
+                    </li>
+                </ol>
+            </nav>
+        )
+    );
+}
+
+function Quicklinks() {
+    const documentData = useContext(DocumentProvider.context);
+    return (
+        documentData && (
+            <div className="quick-links" css={styles.quicklinks}>
+                <div className="quick-links-head">{strings.relatedTopics}</div>
+                <div
+                    dangerouslySetInnerHTML={{
+                        __html: documentData.quickLinksHTML
+                    }}
+                />
+            </div>
+        )
+    );
+}
+
+function Article() {
+    const documentData = useContext(DocumentProvider.context);
+    return (
+        documentData && (
+            <article
+                className="text-content"
+                css={styles.article}
+                dangerouslySetInnerHTML={{ __html: documentData.bodyHTML }}
+            />
+        )
+    );
+}
+
+export default function Page() {
+    return (
+        <div>
+            <Header />
+            <Titlebar />
+            <TOC />
+            <Row css={styles.content}>
+                <Sidebar />
+                <Strut width="3%" />
+                <Article />
+            </Row>
+        </div>
```

Comment:
This is the rough structure of an MDN page, minus the Footer, which I haven't converted to React yet.